rcl_lifecycle/CMakeLists.txt

```diff
@@ -46,8 +46,8 @@ call_for_each_rmw_implementation(targets GENERATE_DEFAULT)
 
 if(BUILD_TESTING)
   if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    set(CMAKE_C_FLAGS ${CMAKE_C_FLAGS} "-std=c11")
-    set(CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} "-std=c++14")
```

Comment:
they are added on the next line through `add_compile_option` to avoid duplication for C and CXX flags